packages/components/src/navigation/level.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * WordPress dependencies
+ */
+import { Icon, chevronLeft } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import { useNavigationContext } from './context';
+import { DEFAULT_LEVEL } from './utils';
+import { BackButtonUI, MenuTitleUI, MenuUI } from './styles/navigation-styles';
+
+export default function NavigationLevel( {
+	children,
+	level = DEFAULT_LEVEL,
+	parentLevel,
+	parentLevelTitle,
```

Comment:
**Note:**
I read through the PR history, but there was quite a bit of discussion 😅 My apologies if this was already covered. (This goes for all of my subsequent comments as well!)

I'm not certain how I feel about the NavigationLevel's knowledge of its parent through the `parentLevel` prop. What do we think about instead passing some sort of onBackButtonClick prop instead?

```javascript
<Navigation>
  <NavigationLevel onBackButtonClick={() => setActiveLevel( 'myParent', 'right' ) }
  </NavigationLevel>
</Navigation>

// level.js
return (
  <div className="className="components-navigation__level">
    { onBackButtonClick && (
      <BackButtonUI onClick={ props.onBackButtonClick }
    ) }
  </div>
)
```

Reply:
This is a bit tricky to explain, especially because I've chosen to "overload" some names in order to make them easier for consumers.

By default, `Navigation` doesn't expose its internal state (active level, active item, and their setters).
Optionally, consumers can provide their own external state to the component (by passing it as `Navigation` props, as you can see in the story example), which will be kept in sync with the internal state.

AFAICS your suggestion would only work if the consumer defines its own external state (`activeLevel`, `setActiveLevel`), but I'd avoid making it required, as I don't see it as a common behaviour.

`parentLevel` would still be required, one way or another (either directly as a prop, or as an argument of `setActiveLevel`), and I think the current solution is easier to use.

`parentLevelTitle` on the other hand is not super useful, as we can infer it by traversing the `children`.
It's a bit of a beefy change, and so I've intentionally kept it out of this PR.
Ideally, the component would figure it out automatically, and we can turn `parentLevelTitle` into an optional `backButtonLabel` instead.

This said, an `onBackButtonClick` prop sounds valuable to me — but for an iteration. 😄